lib/rubygems/util.rb

```diff
@@ -122,7 +122,9 @@ def self.traverse_parents(directory, &block)
 
   def self.glob_files_in_dir(glob, base_path)
     if RUBY_VERSION >= "2.5"
-      Dir.glob(glob, base: base_path).map! {|f| File.join(base_path, f) }
+      Dir.glob(glob, base: base_path).map! do |f|
+        File.expand_path(File.join(base_path, f))
```

Comment:
`File.expand_path(f, base_path)` should work.

Reply:
Thanks for the suggestion, @nobu! This works nicely. Updated.